internal/services/hpccache/hpc_cache_resource.go

```diff
@@ -465,7 +465,7 @@ func resourceHPCCacheCreateOrUpdate(d *pluginsdk.ResourceData, meta interface{})
 			if ad == nil {
 				return fmt.Errorf("Unexpected nil `activeDirectory` in response")
 			}
-			if ad.DomainJoined != storagecache.Yes {
+			if ad.DomainJoined != storagecache.DomainJoinedTypeYes {
```

Comment:
[Old](https://github.com/Azure/azure-sdk-for-go/blob/f3c695847ff5d4c8ee2b432015a2577dff7b1a53/services/storagecache/mgmt/2021-03-01/storagecache/enums.go#L52)
[New](https://github.com/Azure/azure-sdk-for-go/blob/f3c695847ff5d4c8ee2b432015a2577dff7b1a53/services/storagecache/mgmt/2021-09-01/storagecache/enums.go#L56)